server/registry/actions_apis_test.go

```diff
@@ -1153,11 +1154,29 @@ func TestUpdateApiSequence(t *testing.T) {
 	if err := seeder.SeedProjects(ctx, server, seed); err != nil {
 		t.Fatalf("Setup/Seeding: Failed to seed registry: %s", err)
 	}
-	for _, test := range tests {
+	var createTime time.Time
+	var updateTime time.Time
+	for i, test := range tests {
 		t.Run(test.desc, func(t *testing.T) {
-			if _, err := server.UpdateApi(ctx, test.req); status.Code(err) != test.want {
+			var result *rpc.Api
+			var err error
+			if result, err = server.UpdateApi(ctx, test.req); status.Code(err) != test.want {
 				t.Errorf("UpdateApi(%+v) returned status code %q, want %q: %v", test.req, status.Code(err), test.want, err)
 			}
+			if result != nil {
+				if i == 1 {
+					createTime = result.CreateTime.AsTime()
+					updateTime = result.UpdateTime.AsTime()
+				} else {
+					if !createTime.Equal(result.CreateTime.AsTime()) {
+						t.Errorf("UpdateApi create time changed after update (%v %v)", createTime, result.CreateTime.AsTime())
+					}
+					if !updateTime.Before(result.UpdateTime.AsTime()) {
+						t.Errorf("UpdateApi update time did not increase after update (%v %v)", updateTime, result.UpdateTime.AsTime())
+					}
+					updateTime = result.UpdateTime.AsTime()
+				}
+			}
```

Comment:
This seems like it belongs in a separate test. As a rule of thumb, any time you're hiding assertions inside an if-statement to only check them in a subset of the test cases, you're probably trying to squeeze more than one kind of behavioral assertion into a single test.

Here it's really tempting because "TestUpdateApiSequence" is a vague name, where many different tests are probably going to use a sequence of UpdateApi calls. It shouldn't be a catch-all for every type of test that does that though, and should be renamed instead.

Reply:
Ah, ok, I think if you look more closely, you'll see that there's a dependency between these tests - they must run in sequence. I used the existing "test" structure to accomplish this, but could easily rename "test" and "tests" to "step" and "steps", and put the test.Run() outside the loop. WDYT?

Reply:
OK, since these tests will be nonideal without a significant rewrite, I added comments that make their sequential dependency harder to miss and suggest we move forward. We're testing a relatively small feature here, though one that we *have* broken in the past, and I don't think it will be hard to figure out what we've broken if these tests ever fail again. (Happy to add other wording if  you want to more clearly say "these tests are bad, don't do this again!" :) )